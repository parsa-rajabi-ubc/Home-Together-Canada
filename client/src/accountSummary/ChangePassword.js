/**
 * @Author:     Jeff Hatton
 * @Created:    2020.11.19
 *
 * @Description: Change Password input component - returns 3 password text area inputs and a submit button.
 *
 */
import React from 'react';
import PropTypes from "prop-types";
import GenericInput from "../common/forms/GenericInput";

function ChangePassword(props) {
    const {
        oldPassword,
        onOldPasswordChange,
        newPassword,
        onNewPasswordChange,
        confirmedPassword,
        onConfirmedPasswordChange,
        onPasswordChangeSubmit,
        showError,
        errorMessage,
        showSuccess,
        successMessage
    } = props;

    return (
        <div>
            <h3 className={"account-summary-info-header"}> Change Password</h3>
            <p className="account-summary-info-text">
                Remember, your password must be at least 8 characters long and contain a mix of numeric digits, upper
                and lower case letters!
            </p>
            <div className="selected-component-grid-outer">
                <div className="selected-component-grid-inner">
                    <GenericInput
                        className={"input"}
                        classNameLabel={"label"}
                        label={"Old Password "}
                        inputType={"password"}
                        value={oldPassword}
                        onChange={e => onOldPasswordChange(e.target.value)}
                    />
                    <GenericInput
                        className={"input"}
                        classNameLabel={"label"}
                        label={"New Password "}
                        inputType={"password"}
                        value={newPassword}
                        onChange={(e) => {
                            onNewPasswordChange(e.target.value)
                        }}
                    />
                    <GenericInput
                        className={"input"}
                        classNameLabel={"label"}
                        label={"Confirm New Password "}
                        inputType={"password"}
                        value={confirmedPassword}
                        onChange={(e) => {
                            onConfirmedPasswordChange(e.target.value)
                        }}
                    />
                    {showError && <section className={"error-msg mb-4"}>{errorMessage}</section>}
                    {showSuccess && <section className={"success-msg mb-4"}>{successMessage}</section>}
                </div>
            </div>
            <button
                className="btn btn-green form-btn w-1/2"
                onClick={onPasswordChangeSubmit}>
                Update
            </button>
        </div>
    );
}

ChangePassword.propTypes = {
    onPasswordChangeSubmit: PropTypes.func.isRequired,
    onOldPasswordChange: PropTypes.func.isRequired,
    onNewPasswordChange: PropTypes.func.isRequired,
    onConfirmedPasswordChange: PropTypes.func.isRequired,
    showError: PropTypes.bool.isRequired,
    errorMessage: PropTypes.string,
    showSuccess: PropTypes.bool,
    successMessage: PropTypes.string,
    oldPassword: PropTypes.string,
    newPassword: PropTypes.string,
    confirmedPassword: PropTypes.string
};
export default ChangePassword;
