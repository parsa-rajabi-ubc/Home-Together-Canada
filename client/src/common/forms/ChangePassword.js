/**
 * @Author:     Jeff Hatton
 * @Created:    2020.11.19
 *
 * @Description: Change Password input component - returns 3 password text area inputs and a submit button.
 *
 */
import React, {useState} from 'react';
import PropTypes from "prop-types";
import GenericInput from "./GenericInput";

function ChangePassword(props) {
    const {className, onChange, oldPassword, label} = props;
    const [newPassword, setNewPassword] = useState("");
    const [confirmNewPassword, setConfirmNewPassword] = useState("");
    return (
        <label>
            {label}
            <GenericInput className={className} label="Old Password: " type="password" placeholder={oldPassword}/>
            <GenericInput className={className} label="New Password: " inputType="password" onChange={(e)=>{setNewPassword(e.target.value)}}/>
            <GenericInput className={className} label="Confirm New Password: " inputType="password" onChange={(e)=>{setConfirmNewPassword(e.target.value)}}/>
            <GenericInput className={className} label="Submit: " inputType="submit" onClick={onChange}/>
        </label>
    );
}
ChangePassword.propTypes = {
    label: PropTypes.string.isRequired,
    oldPassword: PropTypes.string.isRequired,
    onChange: PropTypes.func.isRequired,
    className: PropTypes.string
};
export default ChangePassword;