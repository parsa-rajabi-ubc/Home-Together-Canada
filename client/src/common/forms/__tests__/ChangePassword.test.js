/**
 * @Author:     Jeff Hatton
 * @Created:    2020.11.21
 *
 * @Description: Change password input Form Component
 *
 */
import React from 'react';
import renderer from  'react-test-renderer'
import ChangePassword from "../ChangePassword";

describe('ConfirmPassword', () => {
    describe('Snapshot test', () => {
        it("should render correctly regardless of properties", () => {
            // given
            const label = 'testLabelString1';
            const onChange = jest.fn();
            const className = 'testClassNameString';
            const oldPassword = 'oldPasswordString1';

            // when
            const component = renderer.create(<ChangePassword className={className} label={label} oldPassword={oldPassword} onChange={onChange}/>).toJSON();

            // then
            expect(component).toMatchSnapshot();
        });
    })
})