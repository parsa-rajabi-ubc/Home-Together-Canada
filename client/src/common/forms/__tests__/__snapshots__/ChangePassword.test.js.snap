// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfirmPassword Snapshot test should render correctly regardless of properties 1`] = `
<label>
  testLabelString1
  <label>
    Old Password: 
    <input
      className="testClassNameString"
      disabled={true}
      placeholder="oldPasswordString1"
      type="password"
    />
  </label>
  <label>
    New Password: 
    <input
      className="testClassNameString"
      disabled={false}
      onChange={[Function]}
      type="password"
    />
  </label>
  <label>
    Confirm New Password: 
    <input
      className="testClassNameString"
      disabled={false}
      onChange={[Function]}
      type="password"
    />
  </label>
  <label>
    Save: 
    <input
      className="testClassNameString"
      disabled={false}
      type="submit"
    />
  </label>
</label>
`;
